### PR TITLE
feat(route): 抖音/头条指数波峰监测

### PR DESCRIPTION
**Pull request by jkker**
## 完整路由地址 / Example for the proposed route(s)

```routes
/oceanengine/教材
/oceanengine/教材/toutiao
```

## 新 RSS 检查列表 / New RSS Script Checklist

- [x] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [x] `Puppeteer`

## 说明 / Note

爬取巨量算数近 6 个月的抖音/头条指数，解密后提取指数波峰当日的热门搜索关键词，并生成为 RSS。主要用于追踪新闻热点事件的后续进展。

### CODE REVIEW

File: lib/routes/oceanengine/index.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+const dayjs = require('dayjs');
+const template = require('art-template');
+template.defaults.minimize = true;
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:3]:
```suggestion
const { art } = require('@/utils/render');
```
You can use RSSHub's wrapper of art-template
https://docs.rsshub.app/joinus/script-standard.html#template-li-zi

File: lib/routes/oceanengine/index.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+dayjs.extend(require('dayjs/plugin/utc'));
+dayjs.extend(require('dayjs/plugin/timezone'));
+const parseDate = (date) => dayjs.tz(date, 'Asia/Shanghai').toDate();
+
+// Parameters
+const CACHE_MAX_AGE = 60 * 60 * 6; // 6 hours
+const DEFAULT_FETCH_DURATION_MONTH = 6; // fetch data 6 month from now
+
+// Fetch Data
+const get_multi_keyword_hot_trend = async (page, keyword, start_date, end_date, app_name = 'aweme') => {
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:13]:
```suggestion
const getMultiKeywordHotTrend = async (page, keyword, start_date, end_date, app_name = 'aweme') => {
```
camelCase

File: lib/routes/oceanengine/index.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+const algorithm = 'aes-128-cfb';
+const decrypt = (rawData) => {
+    const encrypted = Buffer.from(rawData, 'base64').toString('hex');
+    const decipher = createDecipheriv(algorithm, key, iv);
+    const decrypted = decipher.update(encrypted, 'hex', 'utf8') + decipher.final('utf8');
+    return JSON.parse(decrypted);
+};
+
+// Generate Feed
+const search_link_urls = (keyword) => [
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:57]:
```suggestion
const searchLinkUrls = (keyword) => [
```

File: lib/routes/oceanengine/index.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    `https://tophub.today/search?e=tophub&q=${keyword}`,
+    `https://www.baidu.com/s?wd=${keyword}`,
+    `https://www.google.com/search?q=${keyword}`,
+    `https://m.zhihu.com/search?type=content&q=${keyword}`,
+    `https://s.weibo.com/weibo/${keyword}`,
+    `https://www.douyin.com/search/${keyword}`,
+    `https://so.toutiao.com/search?keyword=${keyword}`,
+];
+
+const search_link_names = ['今日热榜', '百度', '谷歌', '知乎', '微博', '抖音', '头条'];
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:67]:
```suggestion
const searchLinkNames = ['今日热榜', '百度', '谷歌', '知乎', '微博', '抖音', '头条'];
```

File: lib/routes/oceanengine/index.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+const create_content = (keyword, queryList, queryListText) =>
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:69]:
```suggestion
const createContent = (keyword, queryList, queryListText) =>
```

File: lib/routes/oceanengine/index.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    template(__dirname + '/item-content.art', {
+        keyword,
+        queryListText,
+        queries: queryList.map((query) => ({
+            links: search_link_urls(encodeURIComponent(query)).map((url, index) => `<a href="${url}" rel="noopener noreferrer" target="_blank">${search_link_names[index]}</a>`),
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:74]:
```suggestion
            links: searchLinkUrls(encodeURIComponent(query)).map((url, index) => `<a href="${url}" rel="noopener noreferrer" target="_blank">${searchLinkNames[index]}</a>`),
```

File: lib/routes/oceanengine/index.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+    const link = `https://trendinsight.oceanengine.com/arithmetic-index/analysis?keyword=${keyword}&appName=${channel}`;
+
+    const item = await ctx.cache.tryGet(
+        link,
+        async () => {
+            const browser = await require('@/utils/puppeteer')();
+            const page = await browser.newPage();
+            await page.goto('https://trendinsight.oceanengine.com/arithmetic-index');
+            const res = await get_multi_keyword_hot_trend(page, keyword, start_date, end_date, channel);
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:102]:
```suggestion
            const res = await getMultiKeywordHotTrend(page, keyword, start_date, end_date, channel);
```

File: lib/routes/oceanengine/index.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+                    title: `${icon} ${keyword} ${relative_percent}% | ${uniqueQueryListStr}`,
+                    author: `巨量算数 - ${channelName}算数指数`,
+                    description: content,
+                    link,
+                    pubDate: parseDate(date),
+                    guid: `巨量算数 - ${channelName}算数指数 | ${keyword} - ${date}`,
+                };
+            });
+        },
+        CACHE_MAX_AGE
```
Comment (TonyRL) [inline lib/routes/oceanengine/index.js:132]: Consider using configurable `contentExpire` from core config
https://github.com/DIYgod/RSSHub/blob/d581d778c693f253b81a71684841af69527b5cce/lib/config.js#L51
```js
const config = require('@/config').value;
// config.cache.contentExpire
```

You may need to pass an extra parameter `false` after cache max age `CACHE_MAX_AGE` for `cache.tryGet(key, getValueFn, maxAge, refresh)` to ensure the cache time will not be extended.

**Review by TonyRL — COMMENTED** [review]
Please follow [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html). New route(s) should be added to lib/v2/*

File: lib/routes/oceanengine/index.js (hunk 8 above)
Comment (Jkker) [reply lib/routes/oceanengine/index.js:132]: The intent here is to extend the cache expire time to 6 hours due to the source website's low update frequency.

Comment (TonyRL) [reply lib/routes/oceanengine/index.js:132]: > due to the source website's low update frequency

That's why `config.cache.contentExpire` existed

`config.cache.routeExpire` defined the cache ttl for frequently updated items.
`config.cache.contentExpire` defined the cache ttl for less-frequent/rarely updated items.

**Comment by Jkker** [comment]
@TonyRL New [PR](https://github.com/DIYgod/RSSHub/pull/9992) created here with all suggestions resolved.

File: lib/routes/oceanengine/index.js (hunk 8 above)
Comment (Jkker) [reply lib/routes/oceanengine/index.js:132]: Thanks. Changes are reflected [here](https://github.com/DIYgod/RSSHub/pull/9992/files#diff-9451ce3b67401576bf943d8de71b3ba08d262d8fdf0651fbb4b396aeded65073R9).

**Pull request closed without merging**